pkg/discovery/dtofactory/container_spec_entity_dto_builder.go

```diff
@@ -100,15 +100,20 @@ func (builder *containerSpecDTOBuilder) getCommoditiesSold(containerSpecMetrics
 		}
 		commSoldBuilder := sdkbuilder.NewCommodityDTOBuilder(commodityType)
 
-		// Aggregate container replicas utilization data.
-		// Note that the returned dataIntervalMs is not the real sampling interval because there could be multiple data
-		// points from container replicas discovered at the same time in one set of data samples. This is just a calculated
-		// equivalent interval between 2 data points to be fed into percentile based algorithm in Turbo server side.
-		// We don't need utilization data for VCPU Throttling and average aggregation strategies are not applicable.
 		if resourceType == metrics.VCPUThrottling {
+			// We don't need utilization data for VCPU Throttling and average aggregation strategies are not applicable.
 			typedUsed, ok := resourceMetrics.Used.([][]metrics.ThrottlingCumulative)
 			if ok {
-				aggregatedCap, aggregatedUsed, aggregatedPeak := aggregateThrottlingSamples(containerSpecMetrics.ContainerSpecId, typedUsed)
+				containerCPUMetrics, exists := containerSpecMetrics.ContainerMetrics[metrics.CPU]
+				if !exists {
+					glog.Errorf("CPU metrics do not exist in resource metrics for ContainerSpec %s", containerSpecMetrics.ContainerSpecId)
+					continue
+				}
+				// Get max container CPU value as containerSpec VCPU capacity which matches the logic in
+				// container_usage_data_aggregator#Aggregate
+				containerSpecVCPUCapacity := aggregation.GetResourceCapacity(containerCPUMetrics)
```

Comment:
Is seems to be always expecting the case where the update (probably by the executed action) has increased the container spec limit. Wouldn't we be actually getting this value wrong in the case where the update has reduced the container spec capacity (say only for example from 2200m to 500m) and we are collecting this data during rolling update.

Would getting the limit (that is the capacity) from the parent spec and then use that as the authoritative capacity (rather then the max) make more sense. Post this any samples which do not match this capacity would be filtered in `filterContainerThrottlingSamples`. 

The change I am suggesting would be in `cluster_monitor.go#genNodePodsMetrics()`, where we would need to get the parent spec to derive capacity(s) from it if they are defined. All specs across all pod replicas will then get same capacity, irresespective of they being older replicas or newer ones. The throttling data from those samples where the capacity does not match this could be discarded.

On the other hand if we don't foresee any implication of using the max always (as is the case in this PR) or possibility of breaking something else by introducing this, we can discard this comment.